gateway-ha/src/test/java/io/trino/gateway/ha/HaGatewayTestUtils.java

```diff
@@ -104,7 +111,9 @@ public static TestConfig buildGatewayConfigAndSeedDb(int routerPort, String conf
                                 "APPLICATION_CONNECTOR_PORT", String.valueOf(30000 + (int) (Math.random() * 1000)))
                         .replace("ADMIN_CONNECTOR_PORT", String.valueOf(31000 + (int) (Math.random() * 1000)))
                         .replace("LOCALHOST_JKS", Paths.get(resource.toURI()).toFile().getAbsolutePath());
-
+        if (extraConfig.isPresent()) {
+            configStr += "\n" + extraConfig.orElseThrow();
+        }
```

Comment:
I think adding a `test-config-without-x-forwarded.yml` might be better than this.